pysatNASA/instruments/icon_mighti.py

```diff
@@ -149,38 +110,166 @@ def init(self):
+            # Find location with any of the flags set
+            _clean_vars(wind_vars, wind_flag, min_val[self.clean_level])
+            _clean_vars(ver_vars, ver_flag, min_val[self.clean_level])
+
+        elif self.tag.find('temp') >= 0:
+            # Neutral temperatures
+            var = 'Temperature'
+            saa_flag = 'Quality_Flag_South_Atlantic_Anomaly'
+            cal_flag = 'Quality_Flag_Bad_Calibration'
+            if saa_flag not in self.variables:
+                id_str = self.inst_id.upper()
+                saa_flag = '_'.join(('ICON_L1_MIGHTI', id_str, saa_flag))
+                cal_flag = '_'.join(('ICON_L1_MIGHTI', id_str, cal_flag))
+                var = '_'.join(('ICON_L23_MIGHTI', id_str, var))
+            # Filter out areas with bad calibration data
+            # as well as data marked in the SAA
+            self[var] = self[var].where((self[saa_flag] == 0)
+                                        & (self[cal_flag] == 0))
+            # Filter out negative temperatures
+            self[var] = self[var].where(self[var] > 0)
 
     return
 
 
-def load(fnames, tag=None, sat_id=None, keep_original_names=False):
+# ----------------------------------------------------------------------------
+# Instrument functions
+#
+# Use the ICON and pysat methods
+
+# Set the list_files routine
+datestr = '{year:04d}{month:02d}{day:02d}_v{version:02d}r{revision:03d}'
+fname1 = 'icon_l2-1_mighti-{id:s}_los-wind-{color:s}_{date:s}.nc'
+fname2 = 'icon_l2-2_mighti_vector-wind-{color:s}_{date:s}.nc'
+fname3 = 'icon_l2-3_mighti-{id:s}_temperature_{date:s}.nc'
+supported_tags = {'': {'vector_wind_green': fname2.format(color='green',
+                                                          date=datestr),
+                       'vector_wind_red': fname2.format(color='red',
+                                                        date=datestr)},
+                  'a': {'los_wind_green': fname1.format(id='a', color='green',
+                                                        date=datestr),
+                        'los_wind_red': fname1.format(id='a', color='red',
+                                                      date=datestr),
+                        'temperature': fname3.format(id='a', date=datestr)},
+                  'b': {'los_wind_green': fname1.format(id='b', color='green',
+                                                        date=datestr),
+                        'los_wind_red': fname1.format(id='b', color='red',
+                                                      date=datestr),
+                        'temperature': fname3.format(id='b', date=datestr)}}
+
+list_files = functools.partial(mm_gen.list_files,
+                               supported_tags=supported_tags)
+
+# Set the download routine
+dirstr1 = '/pub/data/icon/l2/l2-1_mighti-{{id:s}}_los-wind-{color:s}/'
+dirstr2 = '/pub/data/icon/l2/l2-2_mighti_vector-wind-{color:s}/'
+dirstr3 = '/pub/data/icon/l2/l2-3_mighti-{id:s}_temperature/'
```

Comment:
It'd probably be better to have `icon_dir` defined in the icon methods, so if it changes you only have to update it in one place.  But I don't really care if you implement that here or make an issue.

Reply:
Added #76 to address this in future releases.